### PR TITLE
Samiram/about card

### PR DESCRIPTION
**Pull request by samiramkr**
Hi!

During the development of the Pagerduty plugin, I needed something like the AboutCard and IconLinkVertical which was in the Catalog plugin. I created them in Pagerduty but decided to create a general component in the core and use it in Pageduty and Catalog plugins later. 
This PR will remove AboutCard and IconLinkVertical from Catalog and Pagerduty and add those 2 components to the core and use them in the Pagerduty plugin and the Catalog plugin.

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)

### CODE REVIEW

**Comment by benjdlambert** [comment]
@freben / @Rugvip thoughts on this PR? Merge with the understanding that we should break out `core` later on like the avatar?

**Comment by freben** [comment]
Hi! We had a discussion internally about what the core abstraction layers should look like at this point in time.

- Moving `IconLinkVertical` to core makes sense.
- Making a `HeaderIconLinkRow` component makes sense - essentially renaming the `SubHeader` and specialising it.
- And then skip adding the `AboutCard` - it's a high level contract that we may not want to put in place right now.

After this, a user could do

```ts
    <Card>
      <CardHeader
        title={title}
        action={headerAction}
        subheader={<HeaderIconLinkRow links={links} />}
      />
      <Divider />
      <CardContent>
        ...
      </CardContent>
    </Card>
```

I think this makes the icon links contained and composable, without having to make larger scale promises or repeating things that MUI have built in.

Then you could also use the same icon link row in `InfoCard`, or other cards, or other entire contexts.

**Comment by samiramkr** [comment]
> Hi! We had a discussion internally about what the core abstraction layers should look like at this point in time.
> 
> * Moving `IconLinkVertical` to core makes sense.
> * Making a `HeaderIconLinkRow` component makes sense - essentially renaming the `SubHeader` and specialising it.
> * And then skip adding the `AboutCard` - it's a high level contract that we may not want to put in place right now.
> 
> After this, a user could do
> 
> ```ts
>     <Card>
>       <CardHeader
>         title={title}
>         action={headerAction}
>         subheader={<HeaderIconLinkRow links={links} />}
>       />
>       <Divider />
>       <CardContent>
>         ...
>       </CardContent>
>     </Card>
> ```
> 
> I think this makes the icon links contained and composable, without having to make larger scale promises or repeating things that MUI have built in.
> 
> Then you could also use the same icon link row in `InfoCard`, or other cards, or other entire contexts.

It makes sense and I think you are right, it's a better approach to have a HeaderIconLinkRow component that we can use in both InfoCard and AboutCard.

File: packages/core/src/components/HeaderIconLinkRow/types.ts [hunk 1]
```diff
@@ -13,7 +13,12 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-
-export { AboutCard } from './AboutCard';
-// export { SubHeader } from './SubHeader';
-// export { VerticalIcon } from './VerticalIcon';
+export type HeaderIconLink = {
```
Comment (Rugvip) [inline packages/core/src/components/HeaderIconLinkRow/types.ts:16]: Could skip this type and just use `IconLinkVerticalProps` instead right?

File: packages/core/src/components/HeaderIconLinkRow/IconLinkVertical.tsx [hunk 2]
```diff
@@ -67,6 +72,19 @@ export function IconLinkVertical({
...
+        {...props}
+      >
+        {icon}
+        {action}
+      </Link>
+    );
+  }
+
   // Absolute links should not be using RouterLink
   if (href?.startsWith('//') || href?.includes('://')) {
```
Comment (Rugvip) [inline packages/core/src/components/HeaderIconLinkRow/IconLinkVertical.tsx:89]: This should already be handled by https://github.com/backstage/backstage/blob/6a1183a349e0bb06adae4cdadaee9954ce679c77/packages/core/src/components/Link/Link.tsx#L38, so could remove this logic now that we move it to core

File: packages/core/src/components/HeaderIconLinkRow/IconLinkVertical.tsx [hunk 3]
```diff
@@ -13,21 +13,22 @@
...
 import LinkIcon from '@material-ui/icons/Link';
-import { Link as RouterLink } from 'react-router-dom';
+import { Link as RouterLink } from '../Link';
 
-export type IconLinkVerticalProps = {
+type IconLinkVerticalProps = {
   icon?: React.ReactNode;
   href?: string;
   disabled?: boolean;
   title?: string;
```
Comment (Rugvip) [inline packages/core/src/components/HeaderIconLinkRow/IconLinkVertical.tsx:26]: This title doesn't seem to be used much, just passed on to MUI Link which afaik doesn't use it. Could perhaps get rid of it?

File: packages/core/src/components/HeaderIconLinkRow/IconLinkVertical.tsx [hunk 4]
```diff
@@ -67,6 +72,19 @@ export function IconLinkVertical({
     );
   }
 
+  if (action) {
+    return (
+      <Link
+        className={classnames(classes.link, classes.linkStyle)}
+        href={href}
```
Comment (Rugvip) [inline packages/core/src/components/HeaderIconLinkRow/IconLinkVertical.tsx:79]: Action + href feels a bit weird. Could leave this for later rework though

**Review by Rugvip — COMMENTED** [review]
Looks good! 👍 

Couple of nits to clear up before merge

**Review by Rugvip — APPROVED** [review]
👌  :shipit:

**Pull request merged**